### PR TITLE
(Wallet) Remove Android unused code

### PR DESCRIPTION
**Pull request by simoarpe**
## Resolves https://github.com/brave/brave-browser/issues/34526

Removes all unused code caused by transitioning to WebUI.
Removes all unused strings, drawables, and layouts.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [X] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Navigate each and every native screen related to Brave Wallet:
- Wallet creation and restoration
- Unlock screen
- Account selector
- Network Selector
- Transaction bottom sheet fragment
- Transaction advanced options
- Wallet Panel
- DApp panel

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
that's an awesome clean up!

File: android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java [hunk 1]
```diff
@@ -103,14 +97,6 @@ public void resetServices(Context context, TxService mTxService, KeyringService
...
-            mPortfolioModel.resetServices(context, mTxService, mKeyringService, mBlockchainRegistry,
-                    mJsonRpcService, mEthTxManagerProxy, mSolanaTxManagerProxy, mBraveWalletService,
-                    mAssetRatioService);
-            if (mTransactionsModel != null) {
-                mTransactionsModel.resetServices(mContext, mTxService, mKeyringService,
-                        mBlockchainRegistry, mJsonRpcService, mEthTxManagerProxy,
-                        mSolanaTxManagerProxy, mBraveWalletService, mAssetRatioService);
-            }
         }
         init();
```
Comment (supermassive) [inline android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java:101]: These seem no longer needed:
```
mEthTxManagerProxy
mSolanaTxManagerProxy
mAssetRatioService
mBlockchainRegistry
```

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java [hunk 2]
```diff
@@ -188,26 +183,6 @@ public void onCheckedChanged(
...
-                            false, (float) 0.9, true);
-                }
-                if (mType == AdapterType.EDIT_VISIBLE_ASSETS_LIST) {
-                    onWalletListItemClick.onMaybeShowTrashButton(
-                            walletListItemModel, holder.iconTrash);
-                    holder.iconTrash.setOnClickListener(
-                            v -> onWalletListItemClick.onTrashIconClick(walletListItemModel));
-                }
             }
         } else {
```
Comment (supermassive) [inline android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java:187]: WalletCoinAdapter is now never instantiated with these types:
```
VISIBLE_ASSETS_LIST
EDIT_VISIBLE_ASSETS_LIST
```

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailsWithQrActivity.java
Comment (supermassive) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailsWithQrActivity.java:1]: `AccountPrivateKeyActivity` also should be removed now

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java
Comment (supermassive) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java:1]: AddAccountActivity.createIntentToEditAccount no longer needed

File: android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java (hunk 1 above)
Comment (supermassive) [reply android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java:101]: Also `SwapService` no longer needed in native android

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java:1]: Yes, good catch!

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java (hunk 2 above)
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java]: The entire implementation of this adapter was full of old refuse code, unneeded checks and incredibly hard to follow. I've performed a heavy refactor, renamed into a more appropriate name and pruned all the parts that do not make sense anymore. Very tedious (and time-consuming) work, but now it's much more readable. Let me know if you have more suggestions.

File: android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java (hunk 1 above)
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java:101]: Because of the new Brave panel UI redesign these will be most probably re-used very soon. I prefer to leave them for now. Once the redesign is over I'll make sure to remove all the unneeded parts.

**Comment by simoarpe** [comment]
Feedback addressed @supermassive, it seems we have amassed another 800 lines of unused code removed 😁

**Review by petemill — APPROVED** [review]
strings seem all removed, so approving for that

File: android/java/org/chromium/chrome/browser/crypto_wallet/model/AccountSelectorItemModel.java [hunk 3]
```diff
@@ -193,12 +133,12 @@ public String getText2() {
...
-        return mVisible;
+    public boolean isSelected() {
+        return mSelected;
     }
 
-    public void isVisible(final boolean visible) {
-        mVisible = visible;
+    public void isSelected(final boolean selected) {
+        mSelected = selected;
     }
```
Comment (supermassive) [inline android/java/org/chromium/chrome/browser/crypto_wallet/model/AccountSelectorItemModel.java:142]: setSelected?

**Review by supermassive — APPROVED** [review]
lgtm!
